src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -905,7 +893,7 @@ public SolutionState WithProjectReferences(ProjectId projectId, IReadOnlyList<Pr
         /// Creates a new solution instance with the project documents in the order by the specified document ids.
         /// The specified document ids must be the same as what is already in the project; no adding or removing is allowed.
         /// </summary>
-        public SolutionState WithProjectDocumentsOrder(ProjectId projectId, ImmutableList<DocumentId> documentIds)
+        public SolutionState WithProjectDocumentsOrder(ProjectId projectId, ImmutableArray<DocumentId> documentIds)
```

Comment:
I believe `ImmutableList` was intentional, but I don't know if it was because reallocation. I think I was following an existing pattern that was using `ImmutableList`.

Reply:
Reverted to ImmubtaleList.